lib/utils/api_exception_mapper.dart

```diff
@@ -0,0 +1,23 @@
+import 'package:flutter/cupertino.dart';
+import 'package:smoge/app/strings.dart';
+import 'package:smoge/data/api/api_exception.dart';
+
+abstract class ApiExceptionMapper {
+  static String toErrorMessage(Object error, BuildContext context) {
+    if (error is ApiException) {
+      if (error is UnauthorizedException) {
```

Comment:
Nie będzie tu lepszy`switch` ?:)

Reply:
dartowy switch niestety jest biedny i nie obsługuje typów, można by było użyć `runtimeType` w switchu, ale nie jest to najlepsze rozwiązanie (https://github.com/dart-lang/sdk/issues/35052)